modules/images/webp-uploads/load.php

```diff
@@ -683,3 +683,121 @@ function webp_uploads_update_rest_attachment( WP_REST_Response $response, WP_Pos
 	return rest_ensure_response( $data );
 }
 add_filter( 'rest_prepare_attachment', 'webp_uploads_update_rest_attachment', 10, 3 );
+
+/**
+ * Adds sources to metadata for an attachment.
+ *
+ * @param array  $metadata              Metadata of the attachment.
+ * @param array  $valid_mime_transforms List of valid mime transforms for current image mime type.
+ * @param array  $allowed_mimes         Allowed mime types.
+ * @param string $file                  Path to original file.
+ * @return array                        Metadata with sources added.
+ */
+function webp_uploads_update_sources( $metadata, $valid_mime_transforms, $allowed_mimes, $file ) {
+	$original_directory = pathinfo( $file, PATHINFO_DIRNAME );
+
+	foreach ( $valid_mime_transforms as $targeted_mime ) {
+		// Add sources to original image metadata.
+		$extension                  = explode( '|', $allowed_mimes[ $targeted_mime ] );
+		$filename_without_extension = pathinfo( $file, PATHINFO_FILENAME );
+		$image_file                 = path_join( $original_directory, "$filename_without_extension.{$extension[0]}" );
+
+		if ( ! file_exists( $image_file ) ) {
+			continue;
+		}
```

Comment:
See my other comment, we should rather have the _actual_ file name (which is returned by the `WP_Image_Editor::save()` call) passed to this function via parameter.